src/TrashLib/Config/YamlSerializerFactory.cs

```diff
@@ -9,11 +11,13 @@ public class YamlSerializerFactory : IYamlSerializerFactory
 {
     private readonly IObjectFactory _objectFactory;
     private readonly ISecretsProvider _secretsProvider;
+    private readonly IEnvironment _environment;
 
-    public YamlSerializerFactory(IObjectFactory objectFactory, ISecretsProvider secretsProvider)
+    public YamlSerializerFactory(IObjectFactory objectFactory, ISecretsProvider secretsProvider, IEnvironment environment)
```

Comment:
I think this factory is getting a bit too big in terms of responsibility. There's a bit of refactoring I want to do on `master`. Look forward to it. The direction I'd like this class to go:

- It should be unaware of the details (i.e. dependencies) of its type resolvers and node deserializers
- It should not be aware of its node deserializers
- It should consume a collection of node deserializers

I can set up Autofac to register them using reflection and then inject them into this class.

Reply:
@rcdailey  -- not quite sure the emoji for "flying over my head" but I would use it :P  

I think this is what you're talking about: https://fildev.net/2018/07/28/register-dependencies-via-reflection-autofac-module/   Basically there would be some Interface (e.g. `ITrashConfigNodeDeserializer`) and the loader would find every class that implements that interface and pass a collection of those to the `YamlSerializerFactory`.  Each implementation of `ITrashConfigNodeDeserializer` would refer to whatever other Interfaces it needs and AutoFac could do the resolution/dependency injection as it normally does.  Is that the gist?

Reply:
Yes you've got it. I would say that in this case we already have the interface, `INodeDeserializer`, and I'd start with that. There's a possibility that perhaps it's too inclusive... and if that's the case, we could narrow it down using an empty interface like you suggested.